sbws/lib/resultdump.py

```diff
@@ -560,6 +564,6 @@ def results_for_relay(self, relay):
         assert isinstance(relay, Relay)
         fp = relay.fingerprint
         with self.data_lock:
-            if fp not in self.data:
+            if self.data is None or fp not in self.data:
```

Comment:
There are more places where we access self.data than just here. We also access it in self.store_result().

Instead of finding all the places where we use self.data and checking if it is None, I'd prefer either

1. letting the exception happen causing a traceback and sbws to crash. The only time self.data should be None is if it was unable to get initialized with load_recent_results_in_datadir, in which case there's probably something very wrong and we shouldn't try to recover.

2. Initialize self.data to an empty dictionary in `ResultDump.__init__`.

I like option 2 more right now.